index.js

```diff
@@ -348,10 +343,10 @@ var recycleNode = (node) =>
         node
       )
 
-var createVNode = (tag, props, children, type, node) => ({
+var createVNode = (tag, { key, ...props }, children, type, node) => ({
   tag,
   props,
-  key: props.key,
```

Comment:
Nice! I do wonder if destructuring arguments will be "slower" than regular function arguments.

Reply:
I don’t know tbh. I expect it will matter hardly at all because props isn’t monomorphic. But could be worth running the krause bench on this to check.